### PR TITLE
[MaterialCardView] Added options to set custom checkedIcon size and margin

### LINKED ISSUE #1261: [MaterialCardView] Allow custom checked icon size
**Is your feature request related to a problem? Please describe.**
I can't customise the `checkedIcon` size. Currently looks to me is hardcoded to 24dp, which looks great for bigger cards but doesn't look great on cards with small dimensions.

**Describe the solution you'd like**
I'd like to be able to customise the icon size.

**Describe alternatives you've considered**
1. Not using the built in solution and add an ImageView to the MaterialCardView and handle everything myself.
2. Override the `mtrl_card_checked_icon_size` dimensions taking the risk.
`<dimen name="mtrl_card_checked_icon_size" tools:override ="true">18dp</dimen>`

**Additional context**
[Here screenshoots of the default vs desired](https://photos.app.goo.gl/DLdC7oSiPASJLo46A)

### PR DESCRIPTION
**Pull request by gabrielemariotti**
Currently the checked icon has fixed size and margin defined in `dimens.xml` file.

With this PR it is possible to set a custom size and a custom margin.

Schermata 2020-08-13 alle 11 03 01

It closes #1261.

### CODE REVIEW

File: lib/java/com/google/android/material/card/MaterialCardViewHelper.java [hunk 1]
```diff
@@ -165,6 +160,13 @@ void loadFromAttributes(@NonNull TypedArray attributes) {
     setCheckedIcon(
         MaterialResources.getDrawable(
             materialCardView.getContext(), attributes, R.styleable.MaterialCardView_checkedIcon));
+    Resources resources = materialCardView.getResources();
+    setCheckedIconSize(
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardViewHelper.java:164]: set it on the style instead

File: lib/java/com/google/android/material/card/MaterialCardView.java [hunk 2]
```diff
@@ -549,6 +550,70 @@ public void setCheckedIconTint(@Nullable ColorStateList checkedIconTint) {
...
+  public void setCheckedIconSizeResource(@DimenRes int checkedIconSizeResId) {
+    if (checkedIconSizeResId != 0) {
+      cardViewHelper.setCheckedIconSize(getResources().getDimensionPixelSize(checkedIconSizeResId));
+    }
+  }
+
+  /**
+   * Returns the checked icon margin
+   *
+   * @return the checked icon margin
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardView.java:588]: remove this part

File: lib/java/com/google/android/material/card/MaterialCardView.java [hunk 3]
```diff
@@ -549,6 +550,70 @@ public void setCheckedIconTint(@Nullable ColorStateList checkedIconTint) {
     cardViewHelper.setCheckedIconTint(checkedIconTint);
   }
 
+  /**
+   * Returns the checked icon size
+   *
+   * @return the checked icon size
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardView.java:556]: remove

File: lib/java/com/google/android/material/card/res/values/attrs.xml [hunk 4]
```diff
@@ -32,6 +32,10 @@
     <attr name="checkedIcon"/>
     <!-- Tint color for the checked icon. -->
     <attr name="checkedIconTint"/>
+    <!-- Check icon size -->
+    <attr name="checkedIconSize" format="dimension"/>
+    <!-- Check icon margin -->
+    <attr name="checkedIconMargin" format="dimension"/>
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/res/values/attrs.xml:38]: add "for Checkable Cards" to the comments

File: lib/java/com/google/android/material/card/MaterialCardViewHelper.java [hunk 5]
```diff
@@ -397,6 +399,24 @@ void setCheckedIcon(@Nullable Drawable checkedIcon) {
...
+
+  @Dimension
+  public int getCheckedIconMargin() {
+    return checkedIconMargin;
+  }
+
+  public void setCheckedIconMargin(@Dimension int checkedIconMargin) {
+    this.checkedIconMargin = checkedIconMargin;
+  }
+
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardViewHelper.java:419]: make them package private

File: lib/java/com/google/android/material/card/MaterialCardView.java [hunk 6]
```diff
@@ -549,6 +550,70 @@ public void setCheckedIconTint(@Nullable ColorStateList checkedIconTint) {
...
+  }
+
+  /**
+   * Sets the size of the checked icon using a resource id.
+   *
+   * @param checkedIconSizeResId The resource id of this Card's checked icon size
+   * @attr ref com.google.android.material.R.styleable#MaterialCardView_checkedIconSize
+   */
+  public void setCheckedIconSizeResource(@DimenRes int checkedIconSizeResId) {
+    if (checkedIconSizeResId != 0) {
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardView.java:580]: != NO_ID

File: lib/java/com/google/android/material/card/MaterialCardView.java [hunk 7]
```diff
@@ -549,6 +550,70 @@ public void setCheckedIconTint(@Nullable ColorStateList checkedIconTint) {
...
+   */
+  @Dimension
+  public int getCheckedIconMargin() {
+    return cardViewHelper.getCheckedIconMargin();
+  }
+
+  /**
+   * Sets the size of the checked margin
+   *
+   * @param checkedIconMargin checked icon margin
```
Comment (ymarian) [inline lib/java/com/google/android/material/card/MaterialCardView.java:598]: this add anything remove or replace for "the margin in pixels"

File: lib/java/com/google/android/material/card/MaterialCardViewHelper.java (hunk 1 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardViewHelper.java:164]: Do you mean the default value?

Comment (ymarian) [reply lib/java/com/google/android/material/card/MaterialCardViewHelper.java:164]: yeah sorry

yeah sorry that's what I meant

Comment (ymarian) [reply lib/java/com/google/android/material/card/MaterialCardViewHelper.java:164]: fallback value should be 0

File: lib/java/com/google/android/material/card/MaterialCardView.java (hunk 6 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardView.java:575]: Done

File: lib/java/com/google/android/material/card/MaterialCardViewHelper.java (hunk 5 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardViewHelper.java]: Done

File: lib/java/com/google/android/material/card/res/values/attrs.xml (hunk 4 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/res/values/attrs.xml]: Done

File: lib/java/com/google/android/material/card/MaterialCardView.java (hunk 3 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardView.java]: Done

File: lib/java/com/google/android/material/card/MaterialCardView.java (hunk 2 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardView.java]: Done

File: lib/java/com/google/android/material/card/MaterialCardViewHelper.java (hunk 1 above)
Comment (gabrielemariotti) [reply lib/java/com/google/android/material/card/MaterialCardViewHelper.java:163]: Done

**Pull request closed without merging**